src/Microsoft.TemplateEngine.Orchestrator.RunnableProjects/LocalizationModelDeserializer.cs

```diff
@@ -0,0 +1,230 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#nullable enable
+
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.TemplateEngine.Abstractions;
+using Microsoft.TemplateEngine.Orchestrator.RunnableProjects.Localization;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.TemplateEngine.Orchestrator.RunnableProjects
+{
+    internal static class LocalizationModelDeserializer
+    {
+        private const string PostActionIndexPrefix = "postActions[";
+        private const string PostActionIndexSuffix = "]";
+
+        private const string ManualInstructionIndexPrefix = "manualInstructions[";
+        private const string ManualInstructionIndexSuffix = "]";
+
+        public static ILocalizationModel Deserialize(JObject data)
+        {
+            var locModel = new LocalizationModel();
+            var parameterLocalizations = new Dictionary<string, ParameterSymbolLocalizationModel>();
+
+            // Property names are in format: symbols.framework.choices.0.description
+            // Split them using '.' and store together with the localized string (property value).
+            IEnumerable<(IEnumerable<string> nameParts, string localizedString)> stringsWithNames = data.Properties()
+                .Where(p => p.Value.Type == JTokenType.String)
+                .Select(p => (p.Name.Split('.').AsEnumerable(), p.Value.ToString()))
+                .ToList();
+
+            var symbols = LoadSymbolModels(stringsWithNames
+                .Where(s => s.nameParts.FirstOrDefault() == "symbols")
```

Comment:
should constants be extracted and reused in localizer/here?

Reply:
The engine and the localizer doesn't share any dependencies. We could create a new assembly just for these shared strings, but I don't think that will be worth it.